fix(footer): make SVG accessible

shortcuts opened the pull request:
## Summary

After feedbacks from https://twitter.com/SaraSoueidan/status/1491487611831173131, this PR aims at making the footer SVGs accessible.

`keys` or the `Algolia` logo or were not announced by screen readers, this should fix it but I'm not sure if it's the most optimal way to do it.

shortcuts commented on packages/docsearch-react/src/AlgoliaLogo.tsx:
```diff
@@ -18,7 +18,7 @@ export function AlgoliaLogo({ translations = {} }: AlgoliaLogoProps) {
     >
       <span className="DocSearch-Label">{searchByText}</span>
-      <svg width="77" height="19">
+      <svg width="77" height="19" focusable="false" aria-label="Algolia">
```
I've done many tests with the screen reader and when the link is focused it says `Search by Algolia` so I believe it's good

bodinsamuel replied:
ah because it reads the link and the text inside it altogether I guess.
So if the person puts ${searchByText="Use elasticsearch not"} it will read `Use elasticsearch not Algolia` ? ahah

shortcuts commented on packages/docsearch-react/src/Footer.tsx:
```diff
@@ -36,8 +41,12 @@ function CommandIcon(props: CommandIconProps) {
     closeText = 'to close',
+    closeKeyAriaLabel = 'Escape key',
     searchByText = 'Search by',
   } = translations;
```
> It should probably be a sentence like "Press enter to select result"

But then the screen reader reads `Press enter to select result to select`, as it read the whole list item element. This way it's split and make sense but once again I don't know if it's the best way to do it.

I'll educate myself a bit more and read some articles

bodinsamuel replied:
okay yeah I guess same answer than before, if we considers that screen reader reads everything attached that can work. 
I guess one good option is to find big player that does it correctly

SaraSoueidan commented:
- You could have visually-hidden text. Providing alt text (or a label, in this case) for the SVGs is simple, so you could do that using `aria-label`.
- Another option could be something like this (for the Algolia icon):

```html
<svg width="" height="" viewBox="" aria-hidden="false"></svg>
<span class="visually-hidden">Algolia</span>
```

That said, I do prefer adding a label to the SVG in this case to match the visual experience with the SR experience. You'll want to remember to add `role="img"` to the `svg` which has a label.

- You don't need `focusable="false"`. This attribute is useful when you have an SVG inside a button for example, and you want the SVG to not be focusable because if it is then the button will end up having two tab stops. (This was mainly an issue in IE.)

shortcuts commented:
Thanks for the reply and advices @SaraSoueidan! I've actually used your blog article to pick between the multiple solutions :D 

> You'll want to remember to add role="img" to the svg which has a label.

Ok so if I understand correctly, the `role` will only be announced by the SR when the actual element is selected. In that case, when selecting the SVG only and not the list item. Added it!

> You don't need focusable="false"

I've removed it from the logo and it indeed does not changed the SR/tabs behaviors.
